lib/resty/radixtree.lua

```diff
@@ -683,6 +1029,98 @@ function _M.match(self, path, opts)
     return route.metadata
 end
 
+function _M.update_route(self, pre_r, r, opts)
```

Comment:
Please add test cases for new APIs.

Reply:
upper comment contains use cases for each new api. 

--step 2. update route api
local update_opts = {
    id = "00000000000024216171",
    paths = {"/nigolas/cage*", "/tom/cruise"},
    hosts = {"*.angel.city.com", "impossible.mission.com"},
    methods = {"GET", "POST", "PUT"},
    remote_addrs = {"127.0.0.1","192.168.0.0/16",
                    "::1", "fe80::/32"},
    vars = {
        {"arg_name", "==", "cinema"},
        {"arg_weight", "<", 60}, 
    },
    filter_fun = function(vars, opts)
        return vars["arg_name"] == "cinema"
    end,
    metadata = "metadata update route succeed.",
}

rx:update_route(add_opts, update_opts, router_opts)

local opts = {
    host = "impossible.mission.com",
    method = "GET",
    remote_addr = "127.0.0.1",
    vars = ngx.var,
}
ngx.say(rx:match("/nigolas/cage/oscars", opts))

Reply:
Please try to add test cases using test::nginx.